### PR TITLE
Fix console warnings in unit tests

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

This PR cleans up all the unit tests that were causing something to get logged to the console through `console.log`, `console.warn` or `console.error`. 

#### Description of how you validated changes

I ran `yarn test` and confirmed that nothing extraneous is being logged to the console. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/Icon/Icon.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const IconPrimitive: Primitive<IconProps, 'svg'> = (
     _children = children;
   }
   if (paths) {
-    _children = paths.map((path) => <path {...path} />);
+    _children = paths.map((path, index) => <path {...path} key={index} />);
```
Comment (joebuono) [inline packages/react/src/primitives/Icon/Icon.tsx:38]: @zchenwei Did we decide on a better approach than using the `index` as the `key`?

Comment (zchenwei) [reply packages/react/src/primitives/Icon/Icon.tsx:38]: We have a util called `useStableId` that could be used to generate id for `key` prop. However, that util function is dependent on Radix `useId`, which we want to remove it in favor our own implementation. I am not sure how much we progress there.

**Review by zchenwei — APPROVED** [review]
LGTM 🚢

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**